src/form/Switch/Switch.tsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import {
+  Switch as AriaSwitch,
+  SwitchProps as AriaSwitchProps,
+  Label,
+} from "react-aria-components";
+import styled from "styled-components";
+import { theme } from "../../guidelines/theme";
+
+const backgroundColor = "#DADADA";
+const indicatorColor = "#5B51D1";
+const selectedBackgroundColor = "#5B51D1";
+const selectedIndicatorColor = "#DADADA";
+const borderColor = "#979797";
+
+const StyledSwitch = styled(AriaSwitch)<{ isDisabled?: boolean }>`
+  ${({ isDisabled }) =>
+    isDisabled &&
```

Comment:
react-aria-component permet de gérer le disabled via des data attribut sur lesquels on peut s'appuyer pour faire du css plutôt que de faire des tests dans le styled component
```suggestion
    &[data-disabled] {
```
à la place de 
```
${({ isDisabled }) =>
    isDisabled &&
```
Et c'est existant pour les hover, les focus, et autre ...